presto-main/src/main/java/io/prestosql/execution/SqlQueryExecution.java

```diff
@@ -221,9 +226,6 @@ private synchronized void registerDynamicFilteringQuery(PlanRoot plan)
 
     private synchronized void unregisterDynamicFilteringQuery(DynamicFiltersStats finalDynamicFiltersStats)
     {
-        if (!isDone()) {
```

Comment:
Please add `checkState(isDone(), "Expected query to be in done state")` so that the reader knows `unregisterDynamicFilteringQuery` is called from done state. This is important to understand dependencies between `registerDynamicFilteringQuery` and `unregisterDynamicFilteringQuery`